boards/posix/native_posix/sdl_event_loop.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <SDL.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdio.h>
+#include "posix_board_if.h"
+#include "posix_trace.h"
+#include "posix_arch_internal.h"
+#include "soc.h"
+#include "hw_models_top.h"
+
+static void *sdl_event_loop(void *args);
+static void sdl_handle_window_event(const SDL_Event *event);
+static void create_sdl_event_loop_thread(int level);
+
+static pthread_t sdl_event_loop_thread_id;
+static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+
+void sdl_handle_window_event(const SDL_Event *event)
+{
+	SDL_Window *window;
+	SDL_Renderer *renderer;
+
+	switch (event->window.event) {
+	case SDL_WINDOWEVENT_EXPOSED:
+
+		window = SDL_GetWindowFromID(event->window.windowID);
+		if (window == NULL) {
+			return;
+		}
+
+		renderer = SDL_GetRenderer(window);
+		if (renderer == NULL) {
+			return;
+		}
+		SDL_RenderPresent(renderer);
+		break;
+	default:
+		break;
+	}
+}
+
+void *sdl_event_loop(void *args)
+{
+	SDL_Event event;
+
+	if (SDL_Init(SDL_INIT_VIDEO) < 0) {
+		posix_print_error_and_exit("Failed to initialize SDL: %s\n",
+				SDL_GetError());
+	}
+
+	_SAFE_CALL(pthread_mutex_lock(&mutex));
+	_SAFE_CALL(pthread_cond_signal(&cond));
+	_SAFE_CALL(pthread_mutex_unlock(&mutex));
+
+	while (SDL_WaitEvent(&event)) {
+		switch (event.type) {
+		case SDL_WINDOWEVENT:
+			sdl_handle_window_event(&event);
+			break;
+		case SDL_QUIT:
+			hwm_main_loop_exit(0);
+			pthread_exit(NULL);
+			break;
+		default:
+			break;
+		}
+	}
+	fprintf(stderr, "SDL Event loop failed: %s\n", SDL_GetError());
+	hwm_main_loop_exit(1);
+	pthread_exit(NULL);
+}
+
+void create_sdl_event_loop_thread(int level)
+{
+	_SAFE_CALL(pthread_create(&sdl_event_loop_thread_id, NULL,
+				sdl_event_loop, NULL));
+	_SAFE_CALL(pthread_mutex_lock(&mutex));
+	_SAFE_CALL(pthread_cond_wait(&cond, &mutex));
+	_SAFE_CALL(pthread_mutex_unlock(&mutex));
+}
+
+NATIVE_TASK(create_sdl_event_loop_thread, PRE_BOOT_1, 1);
```

Comment:
Minor: You may prefer `PRE_BOOT_2`. It is relatively likely that the user passes some invalid command line arguments to the program, and it exists before reaching `PRE_BOOT_2`, while you don't need to start your SDL thread before that.